### PR TITLE
refactor CorrelationsTwoPoint

### PR DESCRIPTION
**Pull request by yymao**
@vvinuv Here's my attempt to refactor the tpcf test. I think I also fix the bug in the wp(rp).

### CODE REVIEW

**Comment by yymao** [comment]
@vvinuv Once you review and merge this PR in your repo, it will automatically show up in the descqa repo, where I can then merge your PR there.

**Comment by yymao** [comment]
@vvinuv Here's [the web output](https://portal.nersc.gov/project/lsst/descqa/v2/?run=2017-12-21_32) of this PR at 082ab40

**Comment by yymao** [comment]
Updated test run is at https://portal.nersc.gov/project/lsst/descqa/v2/?run=2017-12-21_40 (for this PR at 
9bb8352). w(theta) now looks fine too.

**Comment by yymao** [comment]
@vvinuv this is now ready for review again. Major change since last time is to update the data tables. Also several improvements to the figures. Thanks!

File: descqa/CorrelationsTwoPoint.py [hunk 1]
```diff
@@ -2,314 +2,175 @@
...
+        ))
+        if self.need_distance:
+            filters.extend((
+                '{} < {}'.format(colnames['z'], max(mag_bin['cz_max'] for mag_bin in self.mag_bins)/self._C),
+                '{} >= {}'.format(colnames['z'], min(mag_bin['cz_min'] for mag_bin in self.mag_bins)/self._C),
+            ))
+        catalog_data = catalog_instance.get_quantities(list(colnames.values()), filters=filters)
+        catalog_data = {k: catalog_data[v] for k, v in colnames.items()}
+
+        # create random
```
Comment (vvinuv) [inline descqa/CorrelationsTwoPoint.py:80]: I think some times the sky area is slightly different for objects with different magnitude range. I think generating random catalog once is fine too.

**Review by vvinuv — APPROVED** [review]

**Comment by yymao** [comment]
@vvinuv Thanks. Can you merge this PR here? 

For the footprints used for generating random, they are probably different for observed galaxy sample, but I think they should be the same for mock galaxy catalog? I did try generating randoms separately for individual magnitude cuts but the results look the same.

**Pull request merged**